python/cudf/cudf/core/column/column.py

```diff
@@ -1246,7 +1244,9 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
         if dtype is not None:
             col = col.astype(dtype)
 
-        if np.issubdtype(col.dtype, np.floating):
+        if is_categorical_dtype(col.dtype):
```

Comment:
Since `col` is guaranteed to be a Column here, can just do `isinstance(col, CategoricalColumn)`. Would be cheaper.

Reply:
Nice on Keith, Thanks!